lib/Mojo/WebSocketProxy/Backend/JSONRPC.pm

```diff
@@ -120,8 +122,8 @@ sub call_rpc {
                     if (my $err = $tx->error) {
                         $details .= ', code: ' . ($err->{code} // 'n/a') . ', response: ' . $err->{message};
                     }
-                    warn "WrongResponse [$msg_type], details: $details";
-                    $rpc_failure_cb->($c, $res, $req_storage ) if $rpc_failure_cb;
+                    $log->info("WrongResponse [$msg_type], details: $details");
```

Comment:
The responsibility for logging should move to the failure function.

One option would be to provide the `$log` call as a default, so that the caller can override with something else.